src/plugins/intel_cpu/src/mkldnn_plugin.cpp

```diff
@@ -534,6 +556,75 @@ static void Transformation(CNNNetwork& clonedNetwork, const bool _enableLPT, con
     ConvertToCPUSpecificOpset(nGraphFunc);
 }
 
+
+/**
+ * Map new properties (ov::...) to legacy parameters (CONFIG_KEY(...))
+ *
+ * Direct mapping (no processing required):
+ * ov::enable_profiling.name()       -> CONFIG_KEY(PERF_COUNT)
+ * ov::hint::performance_mode.name() -> CONFIG_KEY(PERFORMANCE_HINT)
+ * ov::hint::num_requests.name()     -> CONFIG_KEY(PERFORMANCE_HINT_NUM_REQUESTS)
+ * ov::cache_dir.name()              -> CONFIG_KEY(CACHE_DIR)
+ *
+ * Indirect mapping (new string name or even different value format):
+ * ov::streams::num.name()              "NUM_STREAMS"           -> CONFIG_KEY(CPU_THROUGHPUT_STREAMS)
+ * ov::affinity.name()                  "AFFINITY"              -> CONFIG_KEY(CPU_BIND_THREAD)
+ * ov::inference_num_threads.name()     "INFERENCE_NUM_THREADS" -> CONFIG_KEY(CPU_THREADS_NUM)
+ * ov::hint::inference_precision.name() "BF16"                  -> CONFIG_KEY(ENFORCE_BF16) "YES"
+ */
+static std::map<std::string, std::string> mapToLegacyConfig(const std::map<std::string, std::string>& config) {
+    const std::unordered_map<std::string, std::string> newToLegacy {
+        {ov::streams::num.name(),              CONFIG_KEY(CPU_THROUGHPUT_STREAMS)},
+        {ov::affinity.name(),                  CONFIG_KEY(CPU_BIND_THREAD)},
+        {ov::inference_num_threads.name(),     CONFIG_KEY(CPU_THREADS_NUM)},
+        {ov::hint::inference_precision.name(), CONFIG_KEY(ENFORCE_BF16)}
+    };
+
+    if (config.count(ov::hint::inference_precision.name())) {
+        const auto& inference_precision = config.at(ov::hint::inference_precision.name());
+        if (!one_of(inference_precision, "bf16", "f32"))
+            IE_THROW() << "Unsupported inference_precision hint: " << inference_precision;
```

Comment:
Can we include `CPU plugin` in the exception message? This will help a user to troubleshoot the problem in case the config is configurated for several plugins.

Reply:
sure, will be added

Reply:
Done